docs/adr/0002-container-cve-remediation.md

```diff
@@ -0,0 +1,34 @@
+
+Red Hat published advisories that describe which images are affected by the found vulnerabilities. This data cannot be used in a straightforward way, so I will describe the evaluation process of a vulnerability to a container image. The advisories contain the publication dates and a list of container images (among other artifacts) affected by the vulnerability. Since images are only uniquely identified by their digest, the list is non-exhaustive and only contains the newest image affected by a vulnerability in a given repo. To figure out if any random image is affected, its age relative to the mentioned image must be examined. In other words, images older than the mentioned one are affected; images newer than the mentioned one have fixed the vulnerability. In other ecosystems this can be accomplished by comparing versions of the packages. In the container ecosystem, the closest analog are tags. However, tags don't have a unified semantic meaning, and floating tags can change its underlying image over time. This means that it's impossible to determine if an image is affected by a vulnerability by comparing its tags. However, each image metadata contains the information of when an image was built, and advisories contain their publication date. With this information, we can claim that images older than an advisory are affected by the vulnerability, and images newer have fixed it. So, if renovate suggests an image bump pull request, if the old image is older and the new image is newer than a vulnerability that affects its repo, we can claim that the PR fixes a vulnerability.
+
+To accommodate this functionality, the images must be specified via their digest. With tags only, it may be impossible to determine vulnerability fixes. For example, if floating tag is used, renovate cannot suggest an update since its underlying digest could have changed silently.
+
+As mentioned previously, renovate has a feature called osvVulnerabilityAlerts. It gets vulnerability information from the database which it downloads from Github during its runtime. The database is compiled from the OSV data twice a day, converted to a renovate-friendly format and released as a Github release. This ensures that it remains up-to-date. The project responsible for releasing and loading the DB is called osv-offline and is one of renovate's dependencies. To implement the new functionality, we need to extend the database with the container and RPM vulnerability data published by Red Hat. This is accomplished by a script in the MintMaker repo, which downloads the latest advisories, converts the data to the OSV format and compiles them to a new database file. It also downloads and extracts the upstream osv-offline database. Our downstream fork of osv-offline is extended to be able to read and search through the new data.
```

Comment:
Could you mention that this is not the official OSV database published by Red Hat, this is an unofficial solution we've created, maybe just as work short-term solution or workaround.